protocol/linkpreview_unfurler_status.go

```diff
@@ -59,7 +59,7 @@ func (u *StatusUnfurler) buildContactData(publicKey string) (*common.StatusConta
 
 	// If no contact found locally, fetch it from waku
 	if contact == nil {
-		if contact, err = u.m.FetchContact(contactID, true); err != nil {
+		if contact, err = u.m.FetchContact(contactID, true); contact == nil || err != nil {
 			return nil, fmt.Errorf("failed to request contact info from mailserver for public key '%s': %w", publicKey, err)
```

Comment:
Just a minor thing: maybe makes sense to separate the checks?
Otherwise when contact not found it will be `failed to request contact info from mailserver for public key '0xdeadbeef': nil` 😄

Reply:
Now should be pretty enough :)